### PR TITLE
Few cleanups

### PR DESCRIPTION
**Pull request by agavignet**
What has been done:
* cleanup unittests: factorize code, use pytest and get rid of TestCase
* improve logging: it is now both printing in stdout and in log file
* introduce `Configuration` class in order to pass parameters in a more elegant way
* suppress some %s

Best review commit by commit

Hope you're doing well dude !

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ language: python
 python:
   - 3.6
 install:
-  - python3 setup.py -q install
+  - pip3 install -r requirements/dev.txt
```
Comment (CedricCabessa) [inline .travis.yml:5]: we need both in order to install "production dependencies" like `appdirs`

File: .travis.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ language: python
...
 script:
   - python3 setup.py flake8
-  - python3 setup.py test
```
Comment (CedricCabessa) [inline .travis.yml:8]: I'd like this to keep working, IMHO it's kind of a standard (no need to know what kind of test software you use)
can you add this in `setup.cfg` ?
```
[aliases]
test=pytest
```

File: README.md [hunk 3]
```diff
@@ -49,13 +49,20 @@ You should edit it to make it more fun :-)
 
 ## Development
 
+### Unittests
+
+```bash
+pip3 install requirements/dev.txt
+pytest -vv -s coffeeconnection/tests
```
Comment (CedricCabessa) [inline README.md:56]: `python setup.py test`

File: requirements/dev.txt [hunk 4]
```diff
@@ -1 +1,2 @@
 black; python_version >= '3.6'
+pytest==5.2.1
```
Comment (CedricCabessa) [inline requirements/dev.txt:2]: There is this big debate about pinning or not.
I'm more in the team "let's not pin and use nighlty build to see what breaks"
But internet says I'm wrong and I admit it is not cool for reproducibility

I might have a look at https://dependabot.com/

We should also pin `appdirs` then.

What are you though on this?

File: coffeeconnection/tests/test_coffeeconnection.py [hunk 5]
```diff
@@ -1,273 +1,269 @@
 import datetime
-import tempfile
-import os
 
-from unittest import TestCase
```
Comment (CedricCabessa) [inline coffeeconnection/tests/test_coffeeconnection.py:5]: Oh did not know pytest was able to works without TestCase! Nice

**Review by CedricCabessa — COMMENTED** [review]
Thanks for all the cleanups!

**Comment by agavignet** [comment]
Hey! I'm off for one week, so review feedbacks will be delayed ;)

File: .travis.yml (hunk 2 above)
Comment (CedricCabessa) [reply .travis.yml:8]: switching to `tox` as the "test" is deprecated https://setuptools.readthedocs.io/en/latest/setuptools.html#test-build-package-and-run-a-unittest-suite
See last commit

**Comment by CedricCabessa** [comment]
I did some changes regarding test, apparently `setup.py test` is deprecated, so I'm switching to tox.
tl;dr; ignore all my comments and rebase :-)
